pandasai/llm/gpt4all.py

```diff
@@ -0,0 +1,105 @@
+"""GPT4All LLMs
+This module provides a family of commercially / non-commercially available 
+LLMs maintained by Nomic AI. With GPT4All we can easily download models like 
+Falcon, Llama etc and run them seamlessly using CPU with lower RAMs. 
+
+Example:
+    Use below example to call GPT4All supporrted models
+
+    >>> from pandasai.llm.gpt4all import GPT4AllLLM
+"""
+import requests
+from typing import Optional
+from gpt4all import GPT4All
+
+from pandasai.prompts.base import Prompt
+from .base import LLM
+
+
+class GPT4AllLLM(LLM):
+    """GPT4All LLMs API
+    Base LLM class is extended to support GPT4AllLLM. When this class will be
+    initialized all the additional parameters like temp, top_p, top_k etc should
+    be inside **kwargs while instantiating the class. That is to be done only if
+    the user wants to override the existing configurations. Below example shows how
+    we can use override certain configurations to change model's behaviour.
+
+    Example:
+        >>> import pandas as pd
+        >>> from pandasai import PandasAI
+        >>> from pandasai.llm.gpt4all import GPT4AllLLM
+
+        >>> model_name = 'ggml-replit-code-v1-3b.bin'
+        >>> model_path = 'path where the model is loaded or to be downloaded'
+        >>> additional_params = {'temp': 1, 'max_tokens': 50}
+        >>> model = GPT4AllLLM(model_folder_path, model_name, **additional_params)
```

Comment:
I believe this is supposed to be 
`model = GPT4AllLLM(model_path, model_name, allow_download=True, **additional_params)`

Notice: here we are generating code and we want the result to be **deterministic** . The temperature should be always set to 0.

Reply:
Yes, that also means that the default params, I will set up the temperature to be 0.